ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2

```diff
@@ -47,6 +47,38 @@ ethernet_interfaces:
 {%                     if adapter_settings.mtu is arista.avd.defined %}
     mtu: {{ adapter_settings.mtu }}
 {%                     endif %}
+{%                     if adapter_settings.port_channel is not arista.avd.defined and adapter_settings.ethernet_segment.short_esi is arista.avd.defined %}
+{%                         if adapter_settings.ethernet_segment.short_esi | lower == 'auto' %}
+{%                             set esi_hash_switches = adapter_settings.switches[:2] | join %}
+{%                             set esi_hash_switch_ports = adapter_settings.switch_ports[:2] | join %}
+{%                             set esi_hash_endpoint_ports = adapter_settings.endpoint_ports[:2] | join %}
+{%                             set esi_hash = (esi_hash_switches ~ esi_hash_switch_ports ~ esi_hash_endpoint_ports ~ channel_group_id) | hash('sha256') %}
+{%                             set short_esi_hash = (esi_hash | regex_replace("([0-9a-f]{4})","\\1:"))[:14] %}
+{%                             do adapter_settings.ethernet_segment.update({ 'short_esi': short_esi_hash }) %}
+{%                         endif %}
+{%                         if adapter_settings.ethernet_segment.short_esi.split(':') | length == 3 %}
+    evpn_ethernet_segment:
+      identifier: {{ adapter_settings.ethernet_segment.short_esi | arista.avd.generate_esi(evpn_short_esi_prefix) }}
+{%                             if adapter_settings.ethernet_segment.redundancy is arista.avd.defined %}
+      redundancy: {{ adapter_settings.ethernet_segment.redundancy }}
+{%                             else %}
+      redundancy: single-active
+{%                             endif %}
+      designated_forwarder_election:
+{%                             if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('modulus') %}
+        algorithm: modulus
+{%                             elif adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('auto') or
+                                    adapter_settings.ethernet_segment.designated_forwarder_preferences is not arista.avd.defined %}
+{%                                 set auto_preferences = range(((adapter_settings.switches | length) * 100) - 100, 0, -100) %}
+        algorithm: preference
+        preference_value: {{ auto_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
+{%                             else %}
+        algorithm: preference
+        preference_value: {{ adapter_settings.ethernet_segment.designated_forwarder_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
+{%                             endif %}
```

Comment:
To change a bit the logic around + adding a check that preferences are a list in the last case (maybe we can even raise a warning if it is not a list?)

PS: this is typed inside github desktop so not 100% sure about indent..

```suggestion
      designated_forwarder_election:
{%                             if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('modulus') %}
        algorithm: modulus
{%                             else %} 
        algorithm: preference
{%                                 if adapter_settings.ethernet_segment.designated_forwarder_preferences is arista.avd.defined('auto') or
                                      adapter_settings.ethernet_segment.designated_forwarder_preferences is not arista.avd.defined %}
{%                                     set auto_preferences = range(((adapter_settings.switches | length) * 100) - 100, 0, -100) %}
        preference_value: {{ auto_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
{%                                 elif adapter_settings.ethernet_segment.designated_forwarder_preferences | arista.avd.defined(var_type='list) %}
        preference_value: {{ adapter_settings.ethernet_segment.designated_forwarder_preferences[adapter_settings.switches.index(inventory_hostname)] | arista.avd.default("0") }}
{%                                 endif %}
{%                             endif %}
```

Reply:
This also makes sense - I just didn't see a nice way to do it; thanks for the var_type tip!  Incorporated in latest commit.